drivers/fuel_gauge/max17048/max17048.c

```diff
@@ -0,0 +1,229 @@
+/* max17048.c - Driver for max17048 battery fuel gauge */
+
+/*
+ * Copyright (c) 2023 Alvaro Garcia Gomez <maxpowel@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT maxim_max17048
+
+#include "max17048.h"
+
+#include <zephyr/drivers/fuel_gauge.h>
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+
+#include <zephyr/logging/log.h>
+
+
+LOG_MODULE_REGISTER(MAX17048);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "MAX17048 driver enabled without any devices"
+#endif
+
+struct max17048_data {
+	float charge;
+	float voltage;
+	uint16_t time_to_full;
+	uint16_t time_to_empty;
+	bool charging;
+};
+
+/* Read buffer */
+unsigned char max17048_buffer[2];
+
+/**
+ * I2C communication
+ * The way we read a value is first writing the address we want to read and then
+ * wait for 2 bytes containing the data.
+ */
+int max17048_read_register(const struct device *dev, uint8_t registerId, uint16_t *response) {
+
+	const struct max17048_config *cfg = dev->config;
+	int rc = i2c_write_read_dt(&cfg->i2c, &registerId, sizeof(registerId), max17048_buffer, sizeof(max17048_buffer));
+	if (rc != 0) {
+		LOG_ERR("Unable to read register, error %d", rc);
+		return rc;
+	}
+	
+	*response = sys_get_be16(max17048_buffer);
+	return 0;
+}
+
+/* Raw value from the internal ADC */
+int max17048_adc(const struct device *i2c_dev, uint16_t *response) {
+	return max17048_read_register(i2c_dev, REGISTER_VCELL, response);
+}
+
+/* Battery voltage */
+int max17048_voltage(const struct device *i2c_dev, float *response) {
+	uint16_t data;
+	int rc = max17048_adc(i2c_dev, &data);
+	if (rc < 0){
+		return rc;
+	}
+	*response = (float)data * 78.125 / 1000000;
+	return 0;
+	
+}
+
+/* Battery percentage */
+int max17048_percent(const struct device *i2c_dev, float *response) {
+	uint16_t data;
+	int rc = max17048_read_register(i2c_dev, REGISTER_SOC, &data);
+	if (rc < 0) {
+		return rc;
+	}
+	*response = (float) data;
+
+	*response /= 256;
+	return 0;
+}
+
+/* % per hour, positive is charging or negative if discharging */
+int max17048_crate(const struct device *i2c_dev, float *response) {
+	int16_t value;
+	int rc = max17048_read_register(i2c_dev, REGISTER_CRATE, &value);
+	if (rc < 0) {
+		return rc;
+	}
+
+	*response = (float)value * 0.208f;
```

Comment:
Can we also document `0.208f`?

Reply:
Again, this weird values just comes from the datasheet
https://www.analog.com/media/en/technical-documentation/data-sheets/MAX17048-MAX17049.pdf
Page 11
Table 2. Register Summary (continued)
0.208%/hr

Sure I will document it.